### PR TITLE
[libc] Allow libc to build on Red Hat

### PR DESCRIPTION
**Pull request by jameshu15869**
Currently, `libc` fails when building on redhat because the triple format uses `redhat` instead of `linux` (The same problem as openSUSE). This PR changes `libc` to accept `redhat` as a valid Linux triple.

### CODE REVIEW

File: libc/cmake/modules/LLVMLibCArchitectures.cmake [hunk 1]
```diff
@@ -167,13 +167,13 @@ if(LIBC_TARGET_OS STREQUAL "baremetal")
   set(LIBC_TARGET_OS_IS_BAREMETAL TRUE)
 elseif(LIBC_TARGET_OS STREQUAL "linux")
   set(LIBC_TARGET_OS_IS_LINUX TRUE)
-elseif(LIBC_TARGET_OS STREQUAL "poky" OR LIBC_TARGET_OS STREQUAL "suse")
-  # poky are ustom Linux-base systems created by yocto. Since these are Linux
+elseif(LIBC_TARGET_OS STREQUAL "poky" OR LIBC_TARGET_OS STREQUAL "suse" OR LIBC_TARGET_OS STREQUAL "redhat")
```
Comment (jhuber6) [inline libc/cmake/modules/LLVMLibCArchitectures.cmake:170]:
```suggestion
elseif(LIBC_TARGET_OS STREQUAL "poky" OR LIBC_TARGET_OS STREQUAL "suse" OR
       LIBC_TARGET_OS STREQUAL "redhat")
```
We at least make an attempt to keep the CMake within 80 columns, sometimes unsuccessfully.

**Review by jhuber6 — COMMENTED** [review]
Thanks for the patch. At this point I'm wondering if we shouldn't have a more intelligent solution for this, but for now it's okay. Just one nit.

**Review by jhuber6 — APPROVED** [review]

File: libc/cmake/modules/LLVMLibCArchitectures.cmake (hunk 1 above)
Comment (jameshu15869) [reply libc/cmake/modules/LLVMLibCArchitectures.cmake]: I tried to run `git clang-format` but it refused to format the file. I saw in `llvm/CMakeLists.txt` that some lines extended past 80 columns, so wasn't sure how to format the CMake file. I'll keep this in mind for the future.

**Comment by jhuber6** [comment]
Thanks, I'll merge it once the CI's done.

**Pull request merged**